### PR TITLE
AAE-19589 Upgrade Java Content SDK to Spring Boot 3.2.1

### PR DESCRIPTION
**Pull request by igdianov**
https://alfresco.atlassian.net/browse/AAE-19589

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ jobs:
     name: "Publish"
     runs-on: ubuntu-latest
     if: >
-      github.event_name == 'push' || contains(github.event.pull_request.labels.*.name, 'preview')
+      github.event_name == 'push'
```
Comment (atchertchian) [inline .github/workflows/ci.yml:64]: The way I understand it, this change makes the "preview" label behave differently on this workflow.
We might want the artifact to be pushed on preview PRs too, no?

Comment (igdianov) [reply .github/workflows/ci.yml:64]: I introduced this tweak to be able to publish preview build artifacts to Nexus, but for some reason, it stopped working, so I changed it back.

**Review by erdemedeiros — APPROVED** [review]
LGTM!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (atchertchian) [reply .github/workflows/ci.yml:64]: ok, I see it at https://github.com/Alfresco/alfresco-java-sdk/pull/213 (for reference)

**Review by atchertchian — APPROVED** [review]

**Pull request merged**